bcda/service/service_test.go

```diff
@@ -362,6 +368,62 @@ func (s *ServiceTestSuite) TestGetNewAndExistingBeneficiaries() {
 	}
 }
 
+// * Live database test *
+//
+// Given the following example scenario:
+// - CCLF File 1 (June)
+// - CCLF File 2 (July)
+// - Request made with "since" parameter later than July
+//
+// We should diff between the correct files (CCLF File 1 and CCLF File 2).
+func (s *ServiceTestSuite) TestGetNewAndExistingBeneficiaries_RecentSinceParameter() {
+	assert := assert.New(s.T())
+	db := database.Connection
+
+	// Test Setup
+	postgrestest.DeleteCCLFFilesByCMSID(s.T(), db, "A0005")
+	defer postgrestest.DeleteCCLFFilesByCMSID(s.T(), db, "A0005")
+
+	acoID := "A0005"
+	cclfFileOld := &models.CCLFFile{CCLFNum: 8, ACOCMSID: acoID, Timestamp: time.Now().Add(-48 * time.Hour), PerformanceYear: 23, Name: "T.BCD.A0005.ZC8Y23.D231119.T1000009", ImportStatus: constants.ImportComplete}
+	cclfFileNew := &models.CCLFFile{CCLFNum: 8, ACOCMSID: acoID, Timestamp: time.Now().Add(-24 * time.Hour), PerformanceYear: 23, Name: "T.BCD.A0005.ZC8Y23.D231120.T1000009", ImportStatus: constants.ImportComplete}
+	postgrestest.CreateCCLFFile(s.T(), db, cclfFileOld)
+	postgrestest.CreateCCLFFile(s.T(), db, cclfFileNew)
+
+	bene1OldRecord := &models.CCLFBeneficiary{FileID: cclfFileOld.ID, MBI: testUtils.RandomMBI(s.T()), BlueButtonID: testUtils.RandomHexID()}
+	bene1NewRecord := &models.CCLFBeneficiary{FileID: cclfFileNew.ID, MBI: bene1OldRecord.MBI, BlueButtonID: testUtils.RandomHexID()}
+	bene2NewRecord := &models.CCLFBeneficiary{FileID: cclfFileNew.ID, MBI: testUtils.RandomMBI(s.T()), BlueButtonID: testUtils.RandomHexID()}
+
+	postgrestest.CreateCCLFBeneficiary(s.T(), db, bene1OldRecord)
+	postgrestest.CreateCCLFBeneficiary(s.T(), db, bene1NewRecord)
+	postgrestest.CreateCCLFBeneficiary(s.T(), db, bene2NewRecord)
```

Comment:
We should really have some sort of [Factory pattern](https://www.stormkit.io/blog/factory-pattern-for-go-tests) for setting up test data so it can look like this:

```
cclfFileOld := Factory.CreateCCLFFile(Timestamp: time.Now().Add(-48 * time.Hour))
cclfFileNew := Factory.CreateCCLFFile(Timestamp: time.Now().Add(-24 * time.Hour))

bene1OldRecord := Factory.CreateCCLFBeneficiary(FileId: cclfFileOld.ID)
bene1NewRecord := Factory.CreateCCLFBeneficiary(FileId: cclfFileNew.ID, MBI: bene1OldRecord.MBI)
bene2NewRecord := Factory.CreateCCLFBeneficiary(FileId: cclfFileNew.ID)
```

Reply:
+1 for the factory pattern. There is a lot of setup in each test, across multiple test files in our repo. I would be really interested in implementing this because it would simplify and clean up a lot of our testing.